smack-experimental/src/main/java/org/jivesoftware/smackx/stanza_content_encryption/element/ContentElement.java

```diff
@@ -0,0 +1,286 @@
+            return this;
+        }
+
+        /**
+         * Set some random length random content padding.
+         *
+         * @return builder
+         */
+        public Builder setRandomPadding() {
+            this.rpad = new RandomPaddingAffixElement();
+            return this;
+        }
+
+        /**
+         * Set the given string as padding.
+         * The padding should be of length between 1 and 200 characters.
+         *
+         * @param padding padding string
+         * @return builder
+         */
+        public Builder setRandomPadding(String padding) {
+            return setRandomPadding(new RandomPaddingAffixElement(padding));
+        }
+
+        /**
+         * Set a padding affix element.
+         *
+         * @param padding affix element
+         * @return builder
+         */
+        public Builder setRandomPadding(RandomPaddingAffixElement padding) {
+            this.rpad = Objects.requireNonNull(padding, "'rpad' affix element MUST NOT be empty.");
+            return this;
+        }
+
+        /**
+         * Add an additional, SCE profile specific affix element.
+         *
+         * @param customAffixElement additional affix element
+         * @return builder
+         */
+        public Builder addFurtherAffixElement(AffixElement customAffixElement) {
+            this.otherAffixElements.add(Objects.requireNonNull(customAffixElement,
+                    "Custom affix element MUST NOT be null."));
+            return this;
+        }
+
+        /**
+         * Add a payload item as child element of the payload element.
+         * There are some items that are not allowed as payload.
+         * Adding those will throw an exception.
+         *
+         * @see <a href="https://xmpp.org/extensions/xep-0420.html#server-processed">
+         *     XEP-0420: Stanza Content Encryption - §9. Server-processed Elements</a>
+         *
+         * @param payloadItem extension element
+         * @return builder
+         * @throws IllegalArgumentException in case an extension element from the blacklist is added.
+         */
+        public Builder addPayloadItem(ExtensionElement payloadItem) {
+            Objects.requireNonNull(payloadItem, "Payload item MUST NOT be null.");
+            this.payloadItems.add(checkForIllegalPayloadsAndPossiblyThrow(payloadItem));
+            return this;
+        }
+
+        /**
+         * Construct a content element from this builder.
+         *
+         * @return content element
+         */
+        public ContentElement build() {
+            List<AffixElement> allAffixElements = collectAffixElements();
+            PayloadElement payloadElement = new PayloadElement(payloadItems);
+            return new ContentElement(payloadElement, allAffixElements);
+        }
+
+        private static ExtensionElement checkForIllegalPayloadsAndPossiblyThrow(ExtensionElement payloadItem) {
+            if (BLACKLISTED_QNAMES.contains(payloadItem.getQName())) {
```

Comment:
nit: I'd prefer having a local variable `QName qname` that is use in `contains()` and in the IAE's message string. Same goes for `String namespace`.